tests/ga/test_extension.py

```diff
@@ -1585,14 +1603,52 @@ def mock_popen(cmd, *args, **kwargs):
                 # The extension status reported by the Handler should be an error since no status file was found
                 self._assert_ext_status(protocol.report_vm_status, ValidHandlerStatus.error, 0,
                                         expected_handler_name="OSTCExtensions.OtherExampleHandlerLinux",
-                                        expected_msg="No such file or directory: '{0}'".format(mock_popen.deleted_status_file))
+                                        expected_msg="No such file or directory: '{0}'".format(
+                                            mock_popen.deleted_status_file))
 
                 # The Handler Status for the dependent extension should be NotReady as it was not executed at all
                 # And since it was not executed, it should not report any extension status either
                 self._assert_handler_status(protocol.report_vm_status, "NotReady", 0, "1.0.0",
                                             expected_msg="Dependent Extension OSTCExtensions.OtherExampleHandlerLinux did not reach a terminal state within the allowed timeout. Last status was {0}".format(
                                                 ValidHandlerStatus.warning))
 
+    def test_it_should_include_part_of_status_in_ext_handler_message(self, mock_http_get, mock_crypt_util, *args):
+        """
+        Testing depends-on scenario when the status file is invalid,
+        The extension status reported by the Handler should contain a fragment of status file for
+        debugging.
+        """
+        exthandlers_handler, protocol = self._create_mock(
+            mockwiredata.WireProtocolData(mockwiredata.DATA_FILE), mock_http_get, mock_crypt_util, *args)
+
+        original_popen = subprocess.Popen
+
+        def mock_popen(cmd, *args, **kwargs):
+            # For the purpose of this test, replacing the status file with file that could not be parsed
+            if "sample.py" in cmd:
+                status_path = os.path.join(kwargs['env'][ExtCommandEnvVariable.ExtensionPath], "status",
+                                           "{0}.status".format(kwargs['env'][ExtCommandEnvVariable.ExtensionSeqNumber]))
+                invalid_json_path = os.path.join(data_dir, "ext", "sample-status-invalid-json-format.json")
+
+                if os.path.exists(status_path):
+                    invalid_json = fileutil.read_file(invalid_json_path)
+                    fileutil.write_file(status_path,invalid_json)
+
+            return original_popen(["echo", "Yes"], *args, **kwargs)
+
+        with patch('azurelinuxagent.common.cgroupapi.subprocess.Popen', side_effect=mock_popen):
+            with patch('azurelinuxagent.ga.exthandlers._DEFAULT_EXT_TIMEOUT_MINUTES', 0.001):
```

Comment:
This patch is not relevant since its not a depends-on case, not needed here.

Reply:
Right, Sorry I missed it . Removed it now.